Objects/longobject.c

```diff
@@ -4317,9 +4241,9 @@ long_true_divide(PyObject *v, PyObject *w)
     */
 
     /* Reduce to case where a and b are both positive. */
-    a_size = Py_ABS(Py_SIZE(a));
-    b_size = Py_ABS(Py_SIZE(b));
-    negate = (Py_SIZE(a) < 0) ^ (Py_SIZE(b) < 0);
+    a_size = _PyLong_DigitCount(a);
+    b_size = _PyLong_DigitCount(b);
+    negate = (_PyLong_IsNegative(a)) ^ (_PyLong_IsNegative(b));
```

Comment:
To avoid UB with bitwise op in signed ints, as we're doing some spring cleaning here:

```suggestion
    negate = _PyLong_IsNegative(a) != _PyLong_IsNegative(b);
```